### PR TITLE
enhance: retry etcd server start with 3 default retries

### PR DESCRIPTION
**Pull request by charleskakumanu**
Added retry method and unit test cases for retrying etcd server start.
New to open source and Go language. Please reject if this is not the requirement/ specify the modifications needed in code.
issue : #17569

### CODE REVIEW

**Comment by amaypatil** [comment]
@xiaofan-luan which release are we targetting this ? I see it was merged with master as well.

**Comment by xiaofan-luan** [comment]
If we cherry pick this to 2.4 branch, we can release at 2.4.8

Other than that it will be 2.5

File: internal/kv/etcd/embed_etcd_kv.go [hunk 1]
```diff
@@ -38,6 +38,11 @@ import (
 // implementation assertion
 var _ kv.MetaKv = (*EmbedEtcdKV)(nil)
 
+const (
+	defaultRetryCount    = 3
```
Comment (dentiny) [inline internal/kv/etcd/embed_etcd_kv.go:42]: I'm wondering if it's better to have a retry library, which allows configuration for:
- max retry count
- max retry timeout
- which error to retry (based a unified error status code)
- enable retry with jitters

Comment (xiaofan-luan) [reply internal/kv/etcd/embed_etcd_kv.go:42]: we do have it with retry.go

Comment (charleskakumanu) [reply internal/kv/etcd/embed_etcd_kv.go:42]: @xiaofan-luan @dentiny 
shall I make the necessary changes and raise a PR?
@amaypatil can you please take this up with me?

Comment (xiaofan-luan) [reply internal/kv/etcd/embed_etcd_kv.go:42]: @charleskakumanu sure, feel free to do sure

**Pull request merged**